File: applicatie/src/main/java/com/vdzon/administratie/model/boekingen/BetalingZonderFactuurBoeking.java
package com.vdzon.administratie.model.boekingen;

import com.vdzon.administratie.model.Boeking;
import com.vdzon.administratie.model.boekingen.relaties.HasAfschrift;
import com.vdzon.administratie.model.boekingen.relaties.HasDeclaratie;
import lombok.*;
import org.mongodb.morphia.annotations.Id;

@ToString
@EqualsAndHashCode
@Getter
@Builder(toBuilder = true)
@AllArgsConstructor(access = AccessLevel.PACKAGE)
@NoArgsConstructor
public class BetalingZonderFactuurBoeking extends Boeking implements HasAfschrift{
    String afschriftUuid;
    @Id
    private String uuid;

    @Override
    public String getAfschriftUuid() {
        return afschriftUuid;
    }
}

File: common/src/main/java/com/vdzon/administratie/model/boekingen/BetalingZonderFactuurBoeking.kt
package com.vdzon.administratie.model.boekingen

import com.vdzon.administratie.model.boekingen.relaties.BoekingMetAfschrift
import org.litote.kmongo.MongoId

class BetalingZonderFactuurBoeking(
        override val afschriftNummer: String,
        @MongoId
        override val uuid: String
) : Boeking(), BoekingMetAfschrift {


    override val omschrijving: String
        get() = "Betaling waar geen rekening van is"

}
